libraries/AP_OSD/AP_OSD_MSP_DisplayPort.cpp

```diff
@@ -77,6 +82,22 @@ void AP_OSD_MSP_DisplayPort::write(uint8_t x, uint8_t y, bool blink, const char
     va_start(ap, fmt);
     int res = hal.util->vsnprintf(buf, sizeof(buf), fmt, ap);
     res = MIN(res, int(sizeof(buf)));
+    const AP_MSP *msp = AP::msp();
+    // Enable packed decimal characters only when using ArduPilot's symbol set
+    // When rendering numbers with a comma (like 0.5) conventional OSD fonts require
+    // 1 character for each digit plus 1 character for the comma.
+    // Packed decimal characters remove the need for the comma character for it's embedded
+    // either in the integer or in the decimal digit i.e. 0.5 requires only 2 digits.
+    if (res > 0 && check_option(AP_OSD::OPTION_DECIMAL_PACK) && msp && !msp->check_option(AP_MSP::MspOption::OPTION_DISPLAYPORT_BTFL_SYMBOLS)) {
```

Comment:
if it's easy to do, then have a common function with AP_OSD_Backend.cpp, or comment saying it is a copy